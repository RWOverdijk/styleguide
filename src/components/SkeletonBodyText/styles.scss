/** @define SkeletonBodyText */

@import "../../css/variables";
@import "../../css/mixins";

.Van-SkeletonBodyText {
  @include skeleton-shimmer;

  backface-visibility: hidden;

  &:not(:first-child) {
    margin-top: 1.2rem;
  }

  &-line {
    @include skeleton-style;

    height: 0.8rem;
    position: relative;
    margin-top: 1.2rem;

    &:first-child {
      margin-top: 0;
    }

    &:last-child:not(:first-child) {
      width: 80%;
    }
  }
}
