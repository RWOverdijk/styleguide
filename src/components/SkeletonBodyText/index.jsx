import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './styles.scss';

class SkeletonBodyText extends PureComponent {
  render() {
    const { className, lines, ...props } = this.props;
    const textLines = [];

    for (let i = 0; i < lines; i += 1) {
      textLines.push(<div className="Van-SkeletonBodyText-line" key={i} />);
    }

    return (
      <div className={classnames('Van-SkeletonBodyText', className)} {...props}>
        {textLines}
      </div>
    );
  }
}

SkeletonBodyText.defaultProps = {
  className: '',
  lines: 3
};

SkeletonBodyText.propTypes = {
  className: PropTypes.string,
  lines: PropTypes.number
};

export default SkeletonBodyText;
