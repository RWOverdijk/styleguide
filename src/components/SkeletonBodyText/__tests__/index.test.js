import React from 'react';
import renderer from 'react-test-renderer';
import SkeletonBodyText from '../';

describe("<SkeletonCard>", () => {
  it("renders without params", () => {
    const tree = renderer.create(
      <SkeletonBodyText />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with 10 lines", () => {
    const tree = renderer.create(
      <SkeletonBodyText lines={10} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <SkeletonBodyText className="my-class" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <SkeletonBodyText data-id="0" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
