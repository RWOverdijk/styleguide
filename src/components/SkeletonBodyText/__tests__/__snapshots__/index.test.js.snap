// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkeletonCard> renders with 10 lines 1`] = `
<div
  className="Van-SkeletonBodyText"
>
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
</div>
`;

exports[`<SkeletonCard> renders with specific attributes 1`] = `
<div
  className="Van-SkeletonBodyText"
  data-id="0"
>
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
</div>
`;

exports[`<SkeletonCard> renders with specific className 1`] = `
<div
  className="Van-SkeletonBodyText my-class"
>
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
</div>
`;

exports[`<SkeletonCard> renders without params 1`] = `
<div
  className="Van-SkeletonBodyText"
>
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
  <div
    className="Van-SkeletonBodyText-line"
  />
</div>
`;
